File: python/leetcode/27-remove-element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        last_index = 0

        for i in range(len(nums)):
            if nums[i] != val:
                if last_index < i:
                    # swap the values:
                    nums[i], nums[last_index] = nums[last_index], nums[i]
                last_index += 1

        return len(nums[:last_index])

File: c#/leetcode/27-remove-element.cs
public class Solution {
    public int RemoveElement(int[] nums, int val) {
        int lastIndex = 0;
        
        for (int i=0; i < nums.Length; i++) {
         
           
            if(nums[i] != val) {
                if (i > lastIndex) {
                    int temp = nums[i];
                    nums[i] = nums[lastIndex];
                    nums[lastIndex] = temp;
                }
                lastIndex++;
            }
        }
        
        return lastIndex;
    }
}